src/libraries/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -165,6 +166,37 @@ public unsafe BitArray(bool[] values)
                     }
                 }
             }
+            else if (AdvSimd.Arm64.IsSupported)
+            {
+                Vector128<byte> elementMask = Vector128.Create(0x01020408_10204080).AsByte();
+                fixed (bool* ptr = values)
+                {
+                    for (; (i + Vector128<byte>.Count * 2) <= values.Length; i += Vector128<byte>.Count * 2)
```

Comment:
Potential integer overflow. If `values.Length` is close to `int.MaxValue`, this loop will never terminate.

Reply:
Wouldn't that be dependent on the runtime changing the current array length cap? (We probably want to fix this anyways, but I don't think it represents a bug today).

Reply:
Yes, but that cap is getting reeeeeally close to `> 0x7FFFFFE0`, at which point overflow is possible. My concern is that it makes this code fragile and prone to failure if that cap is ever lifted.

https://github.com/dotnet/runtime/blob/90ab57c5626df29fbb58e53612defac02833eb8a/src/libraries/System.Private.CoreLib/src/System/Array.cs#L26-L27

Reply:
Do we normally solve this by casting both sides to `uint`?

Reply:
We do the same thing in all of the vectorisation codes elsewhere. I agree we can change the all the loop counters to uint and hopefully eliminate some of that problem...

Reply:
Another reason to make this change: it's not intuitive that this code relies on the max length of a managed array having a safety buffer at the end. If in the future somebody changes the method signature to take span instead of array, they've silently and unwittingly introduced an error into the logic. We had similar issues crop up when converting `int.Parse` to take a span instead of a string: it wasn't obvious that the implementation relied on the string's null terminator, and the change resulted in AVs in some scenarios.

Reply:
Should be addressed now. I made all the loop counters into `uint`s. This should be fine as long as the arrays cannot be longer than `int.MaxValue` (which I highly doubt will change anytime soon).